### PR TITLE
Palworld: allow changing server platform

### PR DESCRIPTION
**Pull request by pbtcameron**
Pull Request Summary:

This pull request enhances the Helm configuration process for PalWorld by adding support for the new AllowConnectPlatform parameter. This parameter is required for configuring Xbox dedicated servers, a feature recently introduced in PalWorld. The changes specifically focus on updating the Helm Questions to include prompts for configuring this new setting, ensuring seamless deployment of PalWorld on Kubernetes with Xbox server support.

Key Changes:

Helm Questions Update: The Helm Questions have been extended to incorporate the new AllowConnectPlatform parameter. This allows users to specify the desired platform connectivity options during Helm chart installation, enabling proper configuration of Xbox dedicated servers.

In reference to [https://github.com/truenas/charts/pull/2795](https://github.com/truenas/charts/pull/2795)

### CODE REVIEW

File: ix-dev/community/palworld/questions.yaml [hunk 1]
```diff
@@ -53,6 +53,15 @@ questions:
...
                   default: ""
+        - variable: allowPlatform
+          label: Platform
+          description: |
+                  Specify which platforms are allowed to connect
+                  Steam or Xbox (Default value is Steam)
+          schema:
+            type: string
+            required: true
+            default: "Steam"
```
Comment (stavros-k) [inline ix-dev/community/palworld/questions.yaml:64]: Lets make this an enum, assuming that its only those 2 values!

File: ix-dev/community/palworld/questions.yaml [hunk 2]
```diff
@@ -53,6 +53,15 @@ questions:
                   type: string
                   private: true
                   default: ""
+        - variable: allowPlatform
+          label: Platform
+          description: |
+                  Specify which platforms are allowed to connect
+                  Steam or Xbox (Default value is Steam)
```
Comment (stavros-k) [inline ix-dev/community/palworld/questions.yaml:60]:
```suggestion
          description: Specify which platforms are allowed to connect
```

Comment (stavros-k) [reply ix-dev/community/palworld/questions.yaml:60]: removed the extra text assuming there will be an enum

File: ix-dev/community/palworld/templates/docker-compose.yaml [hunk 3]
```diff
@@ -130,6 +139,7 @@ services:
         set_ini_value "ServerDescription" "{{ values.palworld.server.description }}" true
         set_ini_value "ServerPassword" '{{ values.palworld.server.password }}' true false
         set_ini_value "AdminPassword" '{{ values.palworld.admin_password }}' true false
+        set_ini_value "AllowConnectPlatform" '{{ values.palworld.allow_platform }}' false true
```
Comment (stavros-k) [inline ix-dev/community/palworld/templates/docker-compose.yaml:142]:
```suggestion
        set_ini_value "AllowConnectPlatform" '{{ values.palworld.allow_platform }}' false
```
print defaults to true anyway!

File: ix-dev/community/palworld/questions.yaml (hunk 1 above)
Comment (pbtcameron) [reply ix-dev/community/palworld/questions.yaml]: Added the Enum

**Review by stavros-k — APPROVED** [review]

**Review by Qubad786 — APPROVED** [review]

**Pull request merged**